wavelength-plugins/psgo.js

```diff
@@ -0,0 +1,545 @@
+
+			open: function (target, room, user) {
+				if (!this.runBroadcast()) return;
+				if (!target) return this.parse(`/help psgo packs open`);
+				if (!Db.userpacks.has(user.userid)) return this.errorReply(`You do not have that pack!`);
+				let index = Db.userpacks.get(user.userid, []).indexOf(target);
+				if (index === -1) return this.errorReply(`You do not have the pack ${target}.`);
+				let array = Db.userpacks.get(user.userid, []);
+				array.splice(index, 1);
+				Db.userpacks.set(user.userid, array);
+				let cards = makePack(target);
+				let results = [];
+				for (let u in cards) {
+					results.push(`<button class="button" name="send" value="/psgo card ${cards[u].id}"><img src="${cards[u].image}" title="${cards[u].id} height="100" width="80"/></button>`);
+				}
+				Db.cards.set(user.userid, Db.cards.get(user.userid, []).concat(cards));
+				return this.sendReplyBox(`You have received the following cards from the ${target} pack:<br/>${results.join('')}`);
+			},
+			openhelp: ['/psgo packs open [pack name] - Open a pack you own.'],
+
+			unopened: 'holding',
+			pending: 'holding',
+			stored: 'holding',
+			holding: function (target, room, user) {
+				if (!Db.userpacks.get(user.userid, []).length) return this.errorReply(`You do not have any packs!`);
+				let usedPacks = {};
+				let userPacks = Db.userpacks.get(user.userid, []);
+				for (let i = 0; i < userPacks.length; i++) {
+					if (!usedPacks[userPacks[i]]) {
+						usedPacks[userPacks[i]] = 1;
+					} else {
+						usedPacks[userPacks[i]]++;
+					}
+				}
+				return this.sendReplyBox(`<strong>Your unopened packs</strong>:<br/>` + Object.keys(usedPacks).map(pack => {
+					return `<button class="button" name="send" value="/psgo packs open ${pack}">Open ${pack} Pack</button> (${usedPacks[pack]} Remaining.)`;
+				}).join('<br/>'));
+			},
+
+			list: function (target, room, user) {
+				if (!this.runBroadcast()) return;
+				this.sendReplyBox(`<div style="max-height: 300px; overflow-y: scroll;"><strong>PSGO Packs</strong><br/><br/>${packs.join(`<br/>`)}</div>`);
+			},
+			listhelp: ['/psgo packs list - displays all psgo packs'],
+		},
+
+		ladder: function (target, room, user) {
+			if (!this.runBroadcast()) return;
+			let values = {Common: 1, Uncommon: 3, Rare: 6, "Ultra Rare": 10, Legendary: 15, Mythic: 20};
+			let keys = Db.cards.keys().map(name => {
+				let points = 0;
+				let userCards = Db.cards.get(name, []);
+				if (userCards.length) {
+					for (let c = 0; c < userCards.length; c++) {
+						points += values[userCards[c].rarity] || 1;
+					}
+				}
+				return {name: name, points: points};
+			});
+			keys.sort(function (a, b) { return b.points - a.points; });
+			return this.sendReplyBox(rankLadder('PSGO Card Ladder', 'Points', keys, 'points'));
+		},
+		ladderhelp: ['/psgo ladder - show the PSGO card point ladder.'],
+
+		nuke: 'reset',
```

Comment:
>nuke

Reply:
Something wrong with this as an alias? or just commenting on it

Reply:
Yeah idk what he means

Reply:
Just commenting